src/extension/background-script/actions/webln/sendPaymentOrPrompt.js

```diff
@@ -0,0 +1,48 @@
+import { parsePaymentRequest } from "invoices";
+import utils from "../../../../common/lib/utils";
+import db from "../../db";
+import sendPayment from "../ln/sendPayment";
+
+const sendPaymentOrPrompt = async (message, sender) => {
+  const paymentRequest = message.args.paymentRequest;
+  const paymentRequestDetails = parsePaymentRequest({
+    request: paymentRequest,
+  });
+
+  const host = message.origin.host;
+  const allowance = await db.allowances
+    .where("host")
+    .equalsIgnoreCase(host)
+    .first();
+
+  if (
+    allowance &&
+    allowance.remainingBudget >= parseInt(paymentRequestDetails.tokens)
+  ) {
+    return sendPaymentWithAllowance(message, paymentRequestDetails, allowance);
+  } else {
+    return payWithPrompt(message, paymentRequestDetails);
+  }
+};
+
+async function sendPaymentWithAllowance(message, paymentRequestDetails) {
+  const response = sendPayment(message);
+  utils.publishPaymentNotification(message, paymentRequestDetails, response);
```

Comment:
It think this should also go into `sendPayment` - we always want to do that when a paument is sent and we have all the data there, don't we?